File: 08FlowControl/for.rs
fn main() {
    // `n` 将在每次迭代中分别取 1, 2, ..., 100
    for n in 1..101 {
    //for n in 1..=100 { //包括边界的写法
        if n % 15 == 0 {
            println!("fizzbuzz");
        } else if n % 3 == 0 {
            println!("fizz");
        } else if n % 5 == 0 {
            println!("buzz");
        } else {
            println!("{}", n);
        }
    }
    
    let names = vec!["Bob", "Frank", "Ferris"];

    for name in names.iter() {
        match name {
            &"Ferris" => println!("There is a rustacean among us!"),
            _ => println!("Hello {}", name),
        }
    }
    
    //move了 所有权发生变化。
    for name in names.into_iter() {
        match name {
            "Ferris" => println!("There is a rustacean among us!"),
            _ => println!("Hello {}", name),
        }
    }
    
    
    //重点的用法，自身引用 使用assign match .. => ...
    let mut names = vec!["Bob", "Frank", "Ferris"];

    for name in names.iter_mut() {
        *name = match name {
            &mut "Ferris" => "There is a rustacean among us!",
            _ => "Hello",
        }
    }
    println!("names: {:?}", names);
}

File: 08FlowControl/for.cs
using System;
public class Program
{
    public static void Main(string[] args)
    {
        Console.WriteLine("This is C# Language");
        // `n` 将在每次迭代中分别取 1, 2, ..., 100
        for(int n=1;n<=100;n++) { //还是简单的for好些，用Enumerable.Range(1, 100) 要引入Linq
            //用?:的嵌套来实现一条语句的写法
            Console.WriteLine(n % 15 == 0 ? "fizzbuzz" : n % 3 == 0 ? "fizz" : n % 5 == 0 ? "buzz" : n.ToString());
        }
        
        string[] names = {"Bob", "Frank", "Ferris"};

        foreach(var name in names ) {
            var message  = name switch{
                "Ferris"=>"There is a rustacean among us!",
                _ => $"Hello {name}",
            };
            Console.WriteLine(message);
        }
        
        //c# 并没有所有权借出以及Move的概念，这里只是增加一些写法罢了。
        for(int i=0; i<names.Length;i++) {
            if( names[i] == "Ferris" )
                Console.WriteLine("There is a rustacean among us!");
            else
                Console.WriteLine($"Hello {names[i]}");
        }
        
        
        //重点的用法，自身引用 使用assign match .. => ...

        foreach(var name in names ) {
            if( name == "Ferris" )
                Console.WriteLine("There is a rustacean among us!");
            else
                Console.WriteLine($"Hello {name}");
        }
        Console.WriteLine("names: {0}", "["+string.Join(",",names)+"]");
    }
}